app/models/queen.rb

```diff
@@ -1,2 +1,21 @@
 class Queen < Piece
+  # Can move horizontal, vertical, or diagonal for unlimited spaces unless path blocked
+  def valid_move?(x,y)
+
+    # Returns false if path is blocked to destiantion space
+      return false if path_blocked?(x,y)
+    # Returns true if moving diagonally)
+    if y == position_y && x == position_x
+      return true if y == x
```

Comment:
Can you walk me through what lines 8 and 9 are meant to do? I have a couple questions but want to make sure I'm thinking about it the same way you are